### ISSUE TITLE
Is NumericVector the right way to go?

### ISSUE THREAD

**Issue by andy-thomason** [extendr/extendr#5]
It may be better to wrap rust **ndarray** instead. We can't make an ndarray owned object, but we can wrap a view or view_mut.

These can be handled by the `export_function` macro,

**Comment by andy-thomason** [comment]
I am currently prototyping a wrapper for **ndarray** in a branch. This also provides error handling for mis-typed parameters.

**Comment by dasmoth** [comment]
I guess you mean the `vector_example` branch here?

I'm currently not having a lot of luck building the example from this.  Fails with:

         cannot find function `unwrap_or_throw` in crate `extendr_api`

There's a crates.io version 0.1.1 of extendr-macros that emits calls to `unwrap_or_throw`, but can't find corresponding source in Github, so suspect there might be an issue with versions getting out of sync here.

**Comment by andy-thomason** [comment]
I'll try to make it work before merging. It may be a version issue.

**Comment by andy-thomason** [comment]
TL;DR is that I don't want to spend time replicating ndarray functionality
and also don't want to be a RCpp clone.

I really like the macros, like rep! as they do not require R users to
retrain
to use Rust.

**Comment by andy-thomason** [comment]
I've merged the prototype work on ndarray integration.
